tests/ct_connection.c

```diff
@@ -434,7 +437,12 @@ void test_async_password_missing(void) {
     assert(acc->err == REDIS_ERR_OTHER);
     assert(strcmp(acc->errstr, "slotmap not available") == 0);
 
-    event_base_dispatch(base);
+    /* Since the owner of the slot is unknown the redisClusterAsyncCommand will
+       initite a slotmap update. This update will register a new socket fd in the
+       event system, which will block libevents API `event_base_dispatch()` until
+       the socket is closed. Use alternative API to run the event loop once only
+       to avoid this problem. */
+    event_base_loop(base, EVLOOP_ONCE);
 
     redisClusterAsyncFree(acc);
```

Comment:
Why run the event loop at all then? Which event is it handling and shouldn't we check afterwards that it did what we expected?

Reply:
Good idea, I have removed the use of an eventloop in this case.